src/main/java/ch/njol/skript/expressions/ExprEntitySound.java

```diff
@@ -0,0 +1,154 @@
+		return true;
+	}
+
+	@Override
+	protected String @Nullable [] get(Event event) {
+		Set<String> sounds = new HashSet<>();
+		for (LivingEntity entity : entities.getArray(event)) {
+			Sound sound = getEntitySound(entity, event);
+			if (sound != null)
+				sounds.add(sound.name());
+		}
+		return sounds.toArray(String[]::new);
+	}
+
+	private @Nullable Sound getEntitySound(LivingEntity entity, Event event) {
+		Sound sound = null;
+		switch (this.sound) {
+			case DAMAGE, DAMAGE + 1 -> sound = entity.getHurtSound();
+			case DEATH, DEATH + 1 -> sound = entity.getDeathSound();
+			case FALL, FALL + 1 -> sound = getFallSound(entity, event);
+			case SWIM, SWIM + 1 -> sound = entity.getSwimSound();
+			case SPLASH, SPLASH + 1 -> sound = big ? entity.getSwimHighSpeedSplashSound() : entity.getSwimSplashSound();
+			case EAT_WITH_ITEM, DRINK_WITH_ITEM -> sound = getConsumeSound(entity, event);
+			case EAT -> sound = entity.getEatingSound(new ItemStack(Material.COOKED_BEEF));
+			case DRINK -> sound = entity.getDrinkingSound(new ItemStack(Material.POTION));
+		}
+		return sound;
+	}
+
+	private Sound getFallSound(LivingEntity entity, Event event) {
+		//noinspection ConstantValue
+		int height = this.height == null ? -1 : this.height.getOptionalSingle(event).orElse(-1).intValue();
+		return height != -1 ? entity.getFallDamageSound(height) :
+			(big ? entity.getFallDamageSoundBig() : entity.getFallDamageSoundSmall());
+	}
+
+	private Sound getConsumeSound(LivingEntity entity, Event event) {
+		ItemStack defaultItem = new ItemStack(sound == EAT_WITH_ITEM ? Material.COOKED_BEEF : Material.POTION);
+		//noinspection ConstantValue
+		ItemStack item = this.item == null ? defaultItem : this.item.getOptionalSingle(event).orElse(new ItemType(defaultItem)).getRandom();
+		if (item == null)
+			item = defaultItem;
+		return sound == EAT_WITH_ITEM ? entity.getEatingSound(item) : entity.getDrinkingSound(item);
+	}
+
+	@Override
+	public boolean isSingle() {
+		return entities.isSingle();
+	}
+
+	@Override
+	public Class<? extends String> getReturnType() {
+		return String.class;
+	}
+
+	@Override
+	@SuppressWarnings("ConstantValue")
+	public String toString(@Nullable Event event, boolean debug) {
+		String name = "unknown";
+		switch (sound) {
+			case DAMAGE, DAMAGE + 1 -> name = "damage";
+			case DEATH, DEATH + 1 -> name = "death";
+			case FALL, FALL + 1 -> {
+				if (this.height == null) {
+					name = big ? "high fall damage" : "normal fall damage";
+				} else {
+					name = "fall damage from a height of " + this.height.toString(event, debug);
+				}
+			}
+			case SWIM, SWIM + 1 -> name = "swim";
+			case SPLASH, SPLASH + 1 -> name = big ? "speedy splash" : "splash";
+			case EAT_WITH_ITEM, DRINK_WITH_ITEM -> name = (sound == EAT_WITH_ITEM ? "eating" : "drinking") + (this.item == null ? " with default item" : " " + this.item.toString(event, debug));
+			case EAT, DRINK -> name = sound == EAT ? "eating" : "drinking";
+		}
```

Comment:
You could use a switch expression to return the string directly rather than reassigning the variable